### ISSUE TITLE
Not receiving a callback from Mythic Apfell.js payload on macOS to the C2

### ISSUE THREAD

**Issue by akshay-1994** [its-a-feature/Mythic#71]
Hi,

I have been testing the Apfell.js payload on my personal macOS machine - 
ProductVersion: 10.15.7
BuildVersion: 19H524

After executing the js with osascript I received the callback first time and then the server crashed. After resetting everything and updating my public IP, I am not able to get a callback anymore? What am I missing and how can I debug. I have double checked the public IP and port, what could be the problem here. Please help me debug.

**Comment by its-a-feature** [comment]
I need a bit more information to help you:
* what version of Mythic are you using (you can see it up in the top right corner)
* when you say the server crashed, what do you mean? 
* when you say you reset everything, what do you mean?
* when you are in Mythic, do you see a red square with a white number in the top right? If so, can you click on it and look to see what the error messages are

**Comment by akshay-1994** [comment]
- Version : v2.1.18
- Server Crash -> I chose Free tier server for the deployment initially. After testing out multiple C2's I installed this one, and while testing my server went down. SSH was not possible, maybe some kernel panic. Then I upgraded the server, so my Public IP got changed. And from then nothing works.
- Reset Everything -> When I first got the agent callback I was happy and File browser impressed me the most. However, After some time the agent hung and didn't respond. No command execution was possible and File browser hung to. That's when I decided to run scripts - reset_database.sh, reset_operation_files.sh, reset_rabbitmq.sh  and setup everything from scratch. Post that no callbacks. I did a fresh install as well after stopping the containers and removing the folder completely. No callback yet.
- Live Event Feed -> No error messages. Only information like Starting HTTP C2 Profile and stuff. My theory is that agent is not able to callback the server. 

Help me debug the issue. I have setup everything as required. But not getting a callback again.

**Comment by its-a-feature** [comment]
Thanks for the info! A few more things:
* When you reset everything, did you generate a new payload or are you trying to use the same payload from before?
* Go to "Global Configurations" -> "C2 Profiles". You should see an entry for the C2 Profile you're using with the payload. Off to the right it'll either say "Internal Server" with a dropdown in red or it'll say "Start Internal Server" in grey. If it's grey, click the button, you should get a notification that the internal server is started again, and if you have agents that are still able to reach you, they should start reaching the Mythic server again.

A note about payloads in Mythic:
Everything is tracked by the database. So, if you created a payload, then cleared your database, when old payloads connect in again, Mythic won't know anything about them, so you'll get a warning notification in your UI and won't be able to task it anymore. So, if you wipe your database with the `reset_database.sh` script, you'll have to regenerate payloads.

A note about the C2 profiles:
When you generate a payload that uses the HTTP profile for example, Mythic will automatically start that c2 profile for you so that your agents can connect in. If you restart mythic though or stop/start the C2 profile containers, the version of Mythic you're using (2.1.18) won't automatically start them back up again. That's why I have that extra step in there for you to go and check that place too.

**Comment by akshay-1994** [comment]
- I have updated the payload to place the new public IP and then downloaded the new Apfell.js file. The browser blocks the file initially, manually intervene to download. Execution with osascript does not give any output. 
- I started all the servers. Created the new payload. No luck!

Mythic:
After wiping I re ran start_mythic.sh which writes the payload to the directory right?

**Comment by its-a-feature** [comment]
One easy way to check for connectivity is to see if you can curl. If you can successfully get through everything with curl (you'll get a warning in the ui about a message Mythic can't parse), then your agent should be able to connect in as well.

You should go to "Create Components" -> "Create Payload" and generate a new payload from scratch with all the right values, then download and test that one. Payloads have embedded UUIDs in them that Mythic uses to track information, so just updating the host information in an old payload won't work because Mythic won't know the UUID that's presented to it.

Mythic:
The `start_mythic.sh` script simply brings all of the containers back online.

**Comment by akshay-1994** [comment]
I got the call back. I found out the issue was that I was testing other C2's too so the apache service which was running was not redirecting the connection to Mythic. After stopping the service and resetting the HTTP C2 container, I was able to get the callback. 

Lesson - 1. Use one C2 at a time.  2. The "View Stdout/Stderr" option in C2 profile is very helpful in debugging. I was lost and then this helped.
